example.php

```diff
@@ -195,4 +195,25 @@ function compareColors(Color $color1, Color $color2)
 
 echo "<h2>Compare colors</h2>";
 compareColors(Color::RED(), Color::BLUE());
-compareColors(Color::YELLOW(), Color::YELLOW());
\ No newline at end of file
+compareColors(Color::YELLOW(), Color::YELLOW());
+
+$serialCopyColor = unserialize(serialize(Color::YELLOW()));
+
+if ($serialCopyColor === Color::YELLOW()) {
+    echo "\nunexected === match on serialized/deserialized value\n";
+}
+else {
+    echo "\n Expected === mismatch on serialized/deserialized value\n";
+}
+
+switch ($serialCopyColor) {
+    case Color::YELLOW():
+        echo "case match!\n";
+        break;
+    case Color::BLACK():
+        echo "case false positive!\n";
+        break;
+    default:
+        echo "case false negative!\n";
+        break;
+}
```

Comment:
expanded example to include a test for expected behavior of serialization/deserialization.